commands/permissions.go

```diff
@@ -109,26 +173,147 @@ func showRoleCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 		return response.Error
 	}
 
-	tpl := `Name: {{.Name}}
-Display Name: {{.DisplayName}}
-Description: {{.Description}}
-Permissions: {{.Permissions}}
-{{range .Permissions}}
-  - {{.}}
-{{end}}
-{{if .BuiltIn}}
-Built in: yes
-{{else}}
-Built in: no
-{{end}}
-{{if .SchemeManaged}}
-Scheme Managed: yes
-{{else}}
-Scheme Managed: no
-{{end}}
-`
-
-	printer.PrintT(tpl, role)
+	sort.Strings(role.Permissions)
+
+	consolePermissionMap := map[string]bool{}
+	for _, perm := range role.Permissions {
+		if strings.HasPrefix(perm, "sysconsole_") {
+			consolePermissionMap[perm] = true
+		}
+	}
+
+	getUsedBy := func(permissionID string) []string {
+		var usedByIDs []string
+		if !strings.HasPrefix(permissionID, "sysconsole_") {
+			usedBy := map[string]bool{} // map to make a unique set
+			for key, vals := range model.SysconsoleAncillaryPermissions {
+				for _, val := range vals {
+					if val.Id == permissionID {
+						if _, ok := consolePermissionMap[key]; ok {
+							usedBy[key] = true
+						}
+					}
+				}
+			}
+			for key := range usedBy {
+				usedByIDs = append(usedByIDs, key)
+			}
+		}
+		return usedByIDs
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', 0)
+
+	// Only show the 3-column view if the role has sysconsole permissions
+	// sysadmin has every permission, so no point in showing the "Used by"
+	// column.
+	if len(consolePermissionMap) > 0 && role.Name != "system_admin" {
+		fmt.Fprintf(w, "Property\tValue\tUsed by\n")
+		fmt.Fprintf(w, "--------\t-----\t-------\n")
+		fmt.Fprintf(w, "Name\t%s\t\n", role.Name)
+		fmt.Fprintf(w, "DisplayName\t%s\t\n", role.DisplayName)
+		fmt.Fprintf(w, "BuiltIn\t%v\t\n", role.BuiltIn)
+		fmt.Fprintf(w, "SchemeManaged\t%v\t\n", role.SchemeManaged)
+		for i, perm := range role.Permissions {
+			if i == 0 {
+				fmt.Fprintf(w, "Permissions\t%s\t%v\n", role.Permissions[0], strings.Join(getUsedBy(role.Permissions[0]), ", "))
+			} else {
+				fmt.Fprintf(w, "\t%s\t%v\n", perm, strings.Join(getUsedBy(perm), ", "))
+			}
+		}
+	} else {
+		fmt.Fprintf(w, "Property\tValue\n")
+		fmt.Fprintf(w, "--------\t-----\n")
+		fmt.Fprintf(w, "Name\t%s\n", role.Name)
+		fmt.Fprintf(w, "DisplayName\t%s\n", role.DisplayName)
+		fmt.Fprintf(w, "BuiltIn\t%v\n", role.BuiltIn)
+		fmt.Fprintf(w, "SchemeManaged\t%v\n", role.SchemeManaged)
+		for i, perm := range role.Permissions {
+			if i == 0 {
+				fmt.Fprintf(w, "Permissions\t%s\n", role.Permissions[0])
+			} else {
+				fmt.Fprintf(w, "\t%s\n", perm)
+			}
+		}
+	}
+
+	w.Flush()
+
+	return nil
+}
+
+func assignUsersCmdF(c client.Client, cmd *cobra.Command, args []string) error {
+	role, response := c.GetRoleByName(args[0])
+	if response.Error != nil {
+		return response.Error
+	}
+
+	for _, username := range args[1:] {
+		user, response := c.GetUserByUsername(username, "")
+		if response.Error != nil {
+			return response.Error
+		}
+
+		startingRoles := strings.Fields(user.Roles)
+		for _, roleName := range startingRoles {
+			if roleName == role.Name {
```

Comment:
This will cause the loop to shortcircuit, so I think it should be:

```suggestion
				continue
```